src/Network/Response.php

```diff
@@ -538,13 +538,18 @@ protected function _sendHeader($name, $value = null)
 
     /**
      * Sends a content string to the client.
+     * If the content is a callable, it is invoked.
      *
      * @param string $content string to send as response body
      * @return void
      */
     protected function _sendContent($content)
     {
-        echo $content;
+        if (!is_string($content) && is_callable($content)) {
+            $content->__invoke();
```

Comment:
@ADmad My fear here is that if you allow a closure to return a value, and then in return you echo that value for them, you actually encourage this behaviour. When in reality, if you're working with streams, you should be outputting them while reading them. However, it can be done both ways, so maybe it's best to set it as your suggesting.

How about something like this:

```php
protected function _sendContent($content)
{
    if ($content instanceof Closure) {
        $content = call_user_func($content);
    }
    echo $content;
}
```